src/menu/columnar_menu.rs

```diff
@@ -822,6 +822,24 @@ mod tests {
             completes_no_shared_prefix: ("all", "all"),
     }
 
+    // https://github.com/nushell/nushell/issues/15535
+    partial_completion_tests! {
+        name: partial_completion_with_quotes,
+        completions: ["`Foo bar`", "`Foo baz`"],
+
+        test_cases:
+            partial_completes_prefix_with_backtick: ("F", "`Foo ba"),
+            partial_completes_case_insensitive: ("foo", "`Foo ba"),
+    }
```

Comment:
@ysthakur the 2nd of [suggested test cases](https://github.com/nushell/reedline/pull/969/files#r2512872770) is just an easier version of this?